test.ts

```diff
@@ -0,0 +1,21 @@
+import { Modem } from './dist';
+
+const myModem = new Modem('COM4', 'SMS', { autoInitOnOpen: false });
+
+myModem.events.on('onWriteToModem', (data) => console.log('Write:', data));
+myModem.events.on('onDataReceived', (data) => console.log('Read:', data));
+
+async function start() {
```

Comment:
With much time we could start trying using Serialport Mock Bindung and simulte the connection for some tests ;-))) but yes ... not thaaat easy

Reply:
These are tests of me, which I have committed in the short time by mistake with. Sorry

Reply:
honestly ... having tests would be awesome ... but also a huuuughe amount of effort ... so would not "bring it in" this PR because there were no tests before :-) but as goal ...

Reply:
Yes, you are right. Tests are something for another pull request. First the goal should be to perform a clean refactor